pkg/authn/identityheaders/identityheaders.go

```diff
@@ -43,6 +43,10 @@ func WithAuthHeaders(handler http.Handler, cfg *AuthnHeaderConfig) http.Handler
 	}
 
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		// We don't want the client to impersonate someone else.
+		req.Header.Del(cfg.UserFieldName)
```

Comment:
this was nearly a CVE ->backport to master

Reply:
It was part of the review I had with @enj. It wouldn't have been forgotten.